MoviesTVSentiments/app/src/androidTest/java/com/google/moviestvsentiments/usecase/assetList/AssetListFragmentTest.java

```diff
@@ -0,0 +1,212 @@
+package com.google.moviestvsentiments.usecase.assetList;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static com.google.moviestvsentiments.assertions.RecyclerViewItemCountAssertion.withItemCount;
+
+import android.content.Intent;
+import android.os.Bundle;
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import androidx.test.core.app.ApplicationProvider;
+import com.google.moviestvsentiments.HiltTestActivity;
+import com.google.moviestvsentiments.R;
+import com.google.moviestvsentiments.di.DatabaseModule;
+import com.google.moviestvsentiments.model.AssetType;
+import com.google.moviestvsentiments.model.SentimentType;
+import com.google.moviestvsentiments.service.database.SentimentsDatabase;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import javax.inject.Inject;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import com.google.moviestvsentiments.HiltFragmentScenario;
+import com.google.moviestvsentiments.usecase.signin.SigninActivity;
+import com.google.moviestvsentiments.util.AssetUtil;
+
+@UninstallModules(DatabaseModule.class)
+@HiltAndroidTest
+public class AssetListFragmentTest {
+
+    private static Bundle createFragmentArgs(SentimentType sentimentType) {
+        Bundle arguments = new Bundle();
+        arguments.putSerializable("sentimentType", sentimentType);
+        return arguments;
+    }
+
+    @Rule
+    public HiltAndroidRule hiltAndroidRule = new HiltAndroidRule(this);
+
+    @Rule
+    public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
+    @Inject
+    SentimentsDatabase database;
+
+    @Before
+    public void setUp() {
+        hiltAndroidRule.inject();
+    }
+
+    @Test
+    public void assetListFragment_startsWithEmptyMovieList() {
```

Comment:
Something you may want to consider here is parameterized testing. I think the test cases are good now, but with parameterized tests, you can get better case coverage with less duplication.

It seems to me https://github.com/Pragmatists/JUnitParams is a good option here, but I am not sure if that test runner is compatible with HiltAndroidTest.

https://drive.google.com/file/d/186vkXaHaWEdWr21eIOSdvGfQiaOmy8Uv/view?usp=sharing
https://drive.google.com/file/d/1pJwYw1oK8U86ZJg4IR3z_PI4FqVFmvQC/view?usp=sharing
Comparison: https://drive.google.com/file/d/1WPqtZFplFHbdoVIlYilGmE7f8QV-CojW/view?usp=sharing